pkg/serverless/invocationlifecycle/init.go

```diff
@@ -143,6 +143,14 @@ func (lp *LifecycleProcessor) initFromSNSEvent(event events.SNSEvent) {
 	lp.requestHandler.event = event
 	lp.addTag(tagFunctionTriggerEventSource, sns)
 	lp.addTag(tagFunctionTriggerEventSourceArn, trigger.ExtractSNSEventArn(event))
+
+	// Check for EventBridge event wrapped by the SNS message
+	var eventBridgeEvent events.EventBridgeEvent
+	if err := json.Unmarshal([]byte(event.Records[0].SNS.Message), &eventBridgeEvent); err == nil {
```

Comment:
As an aside, I wish we didn't have to unmarshal these same event messages twice, once here and once during trace context extraction. Something to think about for a future enhancement.